src/components/AddObsModal.js

```diff
@@ -32,7 +33,17 @@ const AddObsModal = ( { closeModal }: Props ): React.Node => {
     closeModal( );
   };
 
-  const navToPhotoGallery = ( ) => navAndCloseModal( "PhotoGallery" );
+  const navToPhotoGallery = async () => {
```

Comment:
Is there a way to make the navigation experience smoother after closing the image picker? It seems a little glitchy to me that the modal closes and the user sees MyObservations with the AddObsModal still pulled up before navigating to the next screen.

Reply:
Not really sure on this one unless there's an intermediary page that's behind the image picker when it's up.

Also if `AddObsModal`, the call to `selectImagesFromGallery` gets destroyed. 

currently in inat ios, the behavior is similar but when the image picker closes, there is a loading state

Reply:
I agree this isn't ideal, but I think it's fine for now. We could have some empty intermediate as Chris suggested, but I think this is more a question for Abhas when this is ready for him to look at.